internal/invocation/stripe_executor.go

```diff
@@ -0,0 +1,97 @@
+package invocation
+
+import (
+	"fmt"
+	"math/rand"
+	"runtime"
+	"sync"
+)
+
+var (
+	// Default values differ from java impl. Also queue size is calculated differently.
+	// Java Client: queueSize per worker = defaultEventQueueCapacity / defaultEventWorkerCount
+	// Go Client: queueSize per worker = defaultEventQueueCapacity
+	defaultEventQueueCapacity = 10000
+	defaultEventWorkerCount   = runtime.NumCPU()
+)
+
+// executor represents the function that will run on workers of stripeExecutor.
+type executor func(queue chan func(), quit chan struct{}, wg *sync.WaitGroup)
+
+// stripeExecutor executes given "tasks" preserving the order among the ones that are given with the same key.
+type stripeExecutor struct {
+	quit       chan struct{}
+	wg         *sync.WaitGroup
+	execFn     executor
+	taskQueues []chan func()
+	queueCount int
+}
+
+// newStripeExecutor returns a new stripeExecutor with default configuration.
+func newStripeExecutor() stripeExecutor {
+	// ignore error, default values do not raise error.
+	ex, _ := newStripeExecutorWithConfig(defaultEventWorkerCount, defaultEventQueueCapacity)
+	return ex
+}
+
+// newStripeExecutor returns a new stripeExecutor with configured queueCount and queueSize.
+func newStripeExecutorWithConfig(queueCount, queueSize int) (stripeExecutor, error) {
```

Comment:
How about returning only the stripeExecutor? `error` is always nil. We can update it if in the future we need to return an error. Also, should it return `*stripeExecutor`?

Reply:
Got rid off error at https://github.com/hazelcast/hazelcast-go-client/pull/696/commits/67d42e5abc830c5a9b777d23da3aec8bf841e135

Reply:
Refactored stripeExecutor to pointer semantics as you suggested at https://github.com/hazelcast/hazelcast-go-client/pull/696/commits/f8f778f05671ee4f60aeaabd0f062d74352436e9